### PR TITLE
[Feature] Add atomic font tokens

### PR DESCRIPTION
**Pull request by TimRoe**
## Description of Change

Added 6 new sets of tokens forming the atomic basis for fonts:
- family
- letterSpacing
- lineHeight
- paragraphSpacing
- size
- style

Added new token attributes:
- `type` that refines category (e.g. type `family` on category `font`)
- `figma` that is set to true if the token should be part of figma export
- `npm` that is set to true if the token should be part of the NPM/dist export

Updated style-dictionary config/builds:
- Added filters (added note to style-dictionary update ticket that the first 4 may potentially be able to be combined into 1 filter after the update):
   - filter/font/family-npm
   - filter/font/letterSpacing-npm
   - filter/font/size-npm
   - filter/font/lineHeight-npm
   - filter/font-figma
- Removed formats:
   - javascript/es6/vads-colors
   - typescript/es6-declarations/colors
   - javascript/es6/vads-spacing
   - typescript/es6-declarations/spacing
- Added formats:
   - `javascript/es6/simple-key-value`, used by colors/spacing + all new font tokens
      - Should be able to handle all simple key-value pair tokens in the future with only adding a config
   - `typescript/es6-declarations/simple-key-value`, used by colors/spacing + all new font tokens
      - Should be able to handle all simple key-value pair tokens in the future with only adding a config
   - `javascript/es6/fontIndex`, index within js/font folder
   - `typescript/es6-declarations/fontIndex`, index within types/font
- Updated formats: 
   - `json/dtcg` to handle new tokens
   - `javascript/es6/vads-module-export` to be more easily modified in the future by looping through a list of files
   - `typescript/es6-declarations/module` to be more easily modified in the future by looping through a list of files

Figma groupings were split to #495 in hopes of finding a more convenient way to handle them than opting to build the JSON stringwise like we do for other output files.

#### Testing Packages

- [0.17.2-alpha.0](https://www.npmjs.com/package/@department-of-veterans-affairs/mobile-tokens/v/0.17.2-alpha.0)

## Testing

Validated spacing/colors tokens still worked as expected. Validated new font tokens appear to work. Note: Bitter-Regular font family worked in web, but did not work in mobile as we don't have the font included yet (ticket #538).

One issue noted: when typing `font.` the global values documentation was not showing for the various token sets. It was unclear why this is the case since spacing does work and is typed as much the same it can be; additionally CMD + clicking into the set (e.g. font.family) and hovering over the typing there does show the global tooltip doc. Played with this for a bit and was unable to get it to work, but the tooltip doc values on the tokens themselves works fine so this was deemed acceptable since, if anyone is using these directly anyway vs typography tokens, they will almost certainly know which type of font tokens are needed in that line and once they have `font.family.` or similar the tooltip does have the doc of the token value.

- [x] Tested on iOS 
- [x] Tested on Android 
- [x] Tested on Web

## PR Checklist
Code reviewer validation:
- General
	- [ ] PR is linked to ticket(s)
	- [ ] PR has `changelog` label applied if it's to be included in the changelog
	- [ ] Acceptance criteria: 
		- All satisfied _or_
		- Documented reason for not being performed _or_
		- Split to separate ticket and ticket is linked by relevant AC(s)
	- [ ] Above PR sections adequately filled out
	- [ ] If any breaking changes, in accordance with the [pre-1.0.0 versioning guidelines](https://github.com/department-of-veterans-affairs/va-mobile-library#versioning-policy): a CU ticket has been created for the VA Mobile App detailing necessary adjustments with the package version that will be published by this ticket
- Code
	- [ ] Tests are included if appropriate (or split to separate ticket)
	- [ ] New functions have proper TSDoc annotations

## Publish

If changes warrant a new version [per the versioning guidelines](https://github.com/department-of-veterans-affairs/va-mobile-library/blob/main/documentation/versioning.md) and the PR is approved and ready to merge:
- [ ] Merge `main` into branch
- [ ] Merge branch to `main`
- [ ] Verify that [Check Component Integrations](https://github.com/department-of-veterans-affairs/va-mobile-library/actions/workflows/check-component-integrations.yml) workflow ran successfully
- [ ] [Publish new version](https://github.com/department-of-veterans-affairs/va-mobile-library/actions/workflows/publish.yml) after merging

### CODE REVIEW

File: packages/tokens/config.js [hunk 1]
```diff
@@ -47,7 +91,7 @@ module.exports = {
           filter: 'filter/spacing/is-spacing',
         },
         {
-          destination: 'types/theme.d.ts',
+          destination: 'types/themes.d.ts',
```
Comment (TimRoe) [inline packages/tokens/config.js:94]: Renamed to match JS file/export const naming.

**Comment by TimRoe** [comment]
Added @jessicawoodin to the PR to validate the modified tokens match her expectations and no values ended up out of sorts.

**Review by jessicawoodin — APPROVED** [review]
Looks great to me!

File: packages/tokens/build.js [hunk 2]
```diff
@@ -237,28 +340,16 @@ StyleDictionary.registerFormat({
       {},
     )
 
-    // Leave spacing tokens sorted by size
-    if (dictionary.allTokens?.[0].attributes?.category === 'spacing') {
+    const category = dictionary.allTokens?.[0].attributes?.category
+    // Leave spacing and font tokens sorted by source (size)
+    if (category === 'spacing' || category === 'font') {
```
Comment (narin) [inline packages/tokens/build.js:345]: Would it be worth adding a `noSort` option here as well?

**Review by narin — APPROVED** [review]
Some minor suggestions but overall looks good.

Lots of nice improvements to make the formats more generic. Wasn't aware of `options` which makes reusing formats way easier. Great work.

File: packages/tokens/build.js [hunk 3]
```diff
@@ -53,6 +60,37 @@ StyleDictionary.registerFilter({
...
 
+/** Filter to tokens of category 'font', type 'family', and npm true */
+StyleDictionary.registerFilter({
+  name: 'filter/font/family-npm',
+  matcher: (token) => fontFilter(token, 'family'),
+})
+
+/** Filter to tokens of category 'font', type 'letterSpacing', and npm true */
+StyleDictionary.registerFilter({
+  name: 'filter/font/letterSpacing-npm',
```
[the remark this answers is not in the document]
Comment (TimRoe) [reply packages/tokens/build.js]: Updated.

Was naming it this way for the source file name, but looking at it holistically using the type name (kebab to follow the raw token name) probably makes more sense.

File: packages/tokens/build.js [hunk 4]
```diff
@@ -53,6 +60,37 @@ StyleDictionary.registerFilter({
...
+
+/** Filter to tokens of category 'font', type 'size', and npm true */
+StyleDictionary.registerFilter({
+  name: 'filter/font/size-npm',
+  matcher: (token) => fontFilter(token, 'size'),
+})
+
+/** Filter to tokens of category 'font', type 'line-height', and npm true */
+StyleDictionary.registerFilter({
+  name: 'filter/font/lineHeight-npm',
```
[the remark this answers is not in the document]
Comment (TimRoe) [reply packages/tokens/build.js]: Updated.

**Pull request merged**